### ISSUE TITLE
'too many arguments -- limit is 4096' exception

### ISSUE THREAD

**Issue by lazie** [Flinesoft/BartyCrouch#99]
Hi!

I am trying to test in my project and it contains over 7000+ files, after running command like ' **bartycrouch code -p Classes/ -l .** ',  I got this exception: **Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'too many arguments (7102) -- limit is 4096'**. 

 It seems to be an os limit? How can I deal with this situation? Thank you.

**Comment by Dschee** [comment]
Hi there and wow, that's many files! I didn't even know there is a file limit, the big question is where that file limit arises. I should definitively dive into this question some time. But for the time being, an easy solution would be to separate those files into at least two different subfolders and run BartyCrouch twice, once for every subfolder.

**Comment by Dschee** [comment]
@lazie I just validated this issue and it actually happens and can't be fixed very easily. I will try to find a nice solution though. I have a workaround in mind that will probably work, but it's quite hacky, so I'll wait search for a better solution first. I'll keep you updated.

**Comment by Dschee** [comment]
Please note that this issue will be fixed in BartyCrouch 4.1 which is planned soon after the release of BartyCrouch 4.0 which I currently prepare. A beta of 4.0 should be out until tomorrow. While implementing a feature for 4.0 I found an approach that will probably fix this issue once and for all.

**Comment by EternalBlack** [comment]
I currently unfortunately also experience this.
`bartycrouch update -x` give me
```
2019-05-22 09:32:58.741 bartycrouch[47331:688397] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'too many arguments (7167) -- limit is 4096'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff53352cfd __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff7d9fca17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff53352b2f +[NSException raise:format:] + 201
	3   Foundation                          0x00007fff55545619 -[NSConcreteTask launchWithDictionary:error:] + 773
	4   bartycrouch                         0x000000010d719dac $s8SwiftCLI4TaskC6launch030_19E0ACB72F1C972020BFBD69850F9J1FLLyyF + 364
	5   bartycrouch                         0x000000010d719949 $s8SwiftCLI4TaskC7runSyncs5Int32VyF + 25
	6   bartycrouch                         0x000000010d71c9e9 $s8SwiftCLI3run_9arguments9directoryySS_SaySSGSSSgtKFTf4xnn_n + 393
	7   bartycrouch                         0x000000010d718bd9 $s8[1]    47331 abort      bartycrouch update -x
```

`bartycrouch lint -x` gives me
```
Starting Task 'Lint' ...
[1]    47453 illegal hardware instruction  bartycrouch lint -x
```
Currently kinda stuck and had to disable Bartycrouch until this is fixed.

**Comment by Dschee** [comment]
Yeah, this is still planned, the fix isn't trivial though, so it might take a while. Until then instead of disabling BartyCrouch, you could also run it separately for different folders in your project each of which doesn't exceed the limit.

**Comment by Jeehut** [comment]
Hey everyone, I just [released version 4.1.1](https://github.com/Flinesoft/BartyCrouch/releases/tag/4.1.1) of BartyCrouch which attempts to fix this issue. Please try it out and report any issues you might run into. Should be available soon on Homebrew, or if you can't wait, just install it with [Mint](https://github.com/yonaskolb/Mint): `mint install Flinesoft/BartyCrouch`
